src/main/java/uk/gov/hmcts/reform/civil/handler/callback/user/RespondToDefenceSpecCallbackHandler.java

```diff
@@ -563,40 +570,67 @@ private boolean checkPastDateValidation(LocalDate localDate) {
             .build();
     }
 
-    private void buildJudgmentAmountSummaryDetails(CaseData caseData, CaseData.CaseDataBuilder<?, ?> updatedCaseData) {
+    private CallbackResponse buildJudgmentAmountSummaryDetails(CallbackParams callbackParams) {
+        CaseData caseData = callbackParams.getCaseData();
+        CaseData.CaseDataBuilder<?, ?> updatedCaseData = caseData.toBuilder();
 
         BigDecimal claimAmount = caseData.getTotalClaimAmount();
         if (YesOrNo.YES.equals(caseData.getApplicant1AcceptPartAdmitPaymentPlanSpec())) {
             claimAmount = caseData.getRespondToAdmittedClaimOwingAmountPounds();
         }
-        BigDecimal claimFee = MonetaryConversions.penniesToPounds(caseData.getClaimFee().getCalculatedAmountInPence());
-        BigDecimal paidAmount = (caseData.getCcjPaymentPaidSomeOption() == YesOrNo.YES) ? MonetaryConversions.penniesToPounds(
-            caseData.getCcjPaymentPaidSomeAmount()) : ZERO;
-        BigDecimal subTotal = claimAmount.add(claimFee).add(caseData.getTotalInterest());
+        BigDecimal claimFee =  MonetaryConversions.penniesToPounds(caseData.getClaimFee().getCalculatedAmountInPence());
+        BigDecimal paidAmount = (caseData.getCcjPaymentDetails().getCcjPaymentPaidSomeOption() == YesOrNo.YES)
+            ? MonetaryConversions.penniesToPounds(caseData.getCcjPaymentDetails().getCcjPaymentPaidSomeAmount()) : ZERO;
+        BigDecimal fixedCost = setUpFixedCostAmount(claimAmount, caseData);
+        BigDecimal subTotal =  claimAmount.add(claimFee).add(caseData.getTotalInterest()).add(fixedCost);
         BigDecimal finalTotal = subTotal.subtract(paidAmount);
 
-        updatedCaseData.ccjJudgmentAmountClaimAmount(claimAmount);
-        updatedCaseData.ccjJudgmentAmountClaimFee(claimFee);
-        updatedCaseData.ccjJudgmentSummarySubtotalAmount(subTotal);
-        updatedCaseData.ccjJudgmentTotalStillOwed(finalTotal);
-        updatedCaseData.ccjJudgmentAmountInterestToDate(caseData.getTotalInterest());
-        updatedCaseData.ccjPaymentPaidSomeAmountInPounds(paidAmount);
+        CCJPaymentDetails ccjPaymentDetails = CCJPaymentDetails.builder()
+            .ccjJudgmentAmountClaimAmount(claimAmount)
+            .ccjJudgmentAmountClaimFee(claimFee)
+            .ccjJudgmentSummarySubtotalAmount(subTotal)
+            .ccjJudgmentTotalStillOwed(finalTotal)
+            .ccjJudgmentAmountInterestToDate(caseData.getTotalInterest())
+            .ccjPaymentPaidSomeAmountInPounds(paidAmount)
+            .ccjJudgmentFixedCostAmount(fixedCost)
+            .build();
+
+        updatedCaseData.ccjPaymentDetails(ccjPaymentDetails);
+
+        return AboutToStartOrSubmitCallbackResponse.builder()
+            .data(updatedCaseData.build().toMap(objectMapper))
+            .build();
+    }
+
+    private BigDecimal setUpFixedCostAmount(BigDecimal claimAmount, CaseData caseData) {
+        if (!YES.equals(caseData.getCcjPaymentDetails().getCcjJudgmentFixedCostOption())) {
```

Comment:
I am not sure whether this should be in the handler. the handler is big already. and  has too many responsibilities. can we move it somewhere outside the handler and then the handler can use it?

Reply:
Updated